### PR TITLE
ARROW-6286: [GLib] Add support for LargeList type

### PR DESCRIPTION
**Pull request by shiro615**
This PR add support for `GArrowLargeListArrayBuilder`, `GArrowLargeListArray`, and `GArrowLargeListDataType`.

### CODE REVIEW

File: c_glib/arrow-glib/array-builder.cpp [hunk 1]
```diff
@@ -4141,6 +4144,147 @@ garrow_list_array_builder_get_value_builder(GArrowListArrayBuilder *builder)
...
+       GARROW_LARGE_LIST_ARRAY_BUILDER(obj)))
+
+static void
+garrow_large_list_array_builder_dispose(GObject *object)
+{
+  auto priv = GARROW_LARGE_LIST_ARRAY_BUILDER_GET_PRIVATE(object);
+
+  if (priv->value_builder) {
+    auto value_builder_priv = GARROW_ARRAY_BUILDER_GET_PRIVATE(priv->value_builder);
+    value_builder_priv->array_builder = nullptr;
```
Comment (kou) [inline c_glib/arrow-glib/array-builder.cpp]: Do we need them?
They should be done in `priv->value_builder`.

File: c_glib/arrow-glib/composite-array.cpp [hunk 2]
```diff
@@ -151,6 +155,102 @@ garrow_list_array_get_value(GArrowListArray *array,
...
+GArrowArray *
+garrow_large_list_array_get_value(GArrowLargeListArray *array,
+                                  gint64 i)
+{
+  auto arrow_array = garrow_array_get_raw(GARROW_ARRAY(array));
+  auto arrow_large_list_array =
+    static_cast<arrow::LargeListArray *>(arrow_array.get());
+  auto arrow_large_list =
+    arrow_large_list_array->values()->Slice(arrow_large_list_array->value_offset(i),
+                                            arrow_large_list_array->value_length(i));
```
Comment (kou) [inline c_glib/arrow-glib/composite-array.cpp]: Could you use `value_slice()` instead of `values()->Slice()`?

**Review by kou — CHANGES_REQUESTED** [review]
Thanks!
Could you check my comments?

**Comment by kou** [comment]
Could you also look into the Travis CI failure on macOS?
(It's not related to this change.)

File: c_glib/arrow-glib/composite-array.cpp (hunk 2 above)
Comment (shiro615) [reply c_glib/arrow-glib/composite-array.cpp]: I've fixed this.

**Comment by shiro615** [comment]
Sorry, I didn't confirm the CI result. I've rebased this branch on master.
I will confirm the CI result later.

File: c_glib/arrow-glib/array-builder.cpp (hunk 1 above)
Comment (shiro615) [reply c_glib/arrow-glib/array-builder.cpp]: I've tried to remove `value_builder_priv->array_builder = nullptr`. 
It seem to be `Segmentation fault` in `garrow_array_builder_finalize()` when running the test of large list array builder. 
I don't know how to resolve this...

Comment (kou) [reply c_glib/arrow-glib/array-builder.cpp]: OK. I've implemented.
We need to use `garrow_array_builder_release_ownership(priv->value_builder)`.

**Review by kou — APPROVED** [review]
+1
We can merge this once CI is green except known CI failures. 

Travis CI failure may be related to https://github.com/protocolbuffers/protobuf/issues/6619

See also: https://lists.apache.org/thread.html/c33a9a7769e73f189ebb05e134c3cc7d1e96ccb09d4457cb28b49142@%3Cdev.arrow.apache.org%3E

So we don't need to look into this.

**Pull request closed without merging**